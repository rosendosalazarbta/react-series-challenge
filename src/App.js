import React, { Component } from 'react';
import { Link, Redirect, Route, Switch, } from 'react-router-dom';
import ComponentHome from './components/ComponentHome';
import ComponentFavorites from './components/ComponentFavorites';
import './styles/challenge.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <header className="navbar">
          <div className="navbar-container">
            <ul>
              <li>
                <Route path="/Home" children={({ match }) => {
                  const active = match ? 'active' : false;
                  return <Link to="/Home" className={active}>Home</Link>;
                }} />
              </li>
              <li>
                <Route path="/Favorites" children={({ match }) => {
                  const active = match ? 'active' : false;
                  return <Link to="/Favorites" className={active}>Favorites</Link>;
                }} />
              </li>
            </ul>
          </div>
        </header>
        <Switch>
          <Route path="/Home" component={ComponentHome} />
          <Route path="/Favorites" component={ComponentFavorites} />
          <Redirect to={{ pathname: "/Home", }} />
        </Switch>
      </div >
    );
  }
}

export default App;
