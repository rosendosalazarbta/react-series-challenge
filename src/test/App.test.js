import App from '../App';
import React from 'react';
import { shallow } from 'enzyme';
import { Navbar, NavbarContainer, NavbarLi, NavbarLink, AppBody, } from '../styles/App.style';
import ErrorBoundary from '../components/ErrorBoundary';
import { Switch } from 'react-router-dom';

describe('App', () => {
  let Wrapper;

  beforeEach(() => {
    Wrapper = shallow(<App />);
  });

  test('Should match snapshot', () => {
      expect(Wrapper).toMatchSnapshot();
  });

  test('Should render Navbar', () => {
    expect(Wrapper.find(Navbar)).toHaveLength(1);
  });

  test('Should render NavbarContainer', () => {
    expect(Wrapper.find(NavbarContainer)).toHaveLength(1);
  });

  test('Should render NavbarLi', () => {
    expect(Wrapper.find(NavbarLi)).toHaveLength(2);
  });

  test('Should render NavbarLink', () => {
    expect(Wrapper.find(NavbarLink)).toHaveLength(0);
  });

  test('Should render AppBody', () => {
    expect(Wrapper.find(AppBody)).toHaveLength(1);
  });

  test('Should render ErrorBoundary', () => {
    expect(Wrapper.find(ErrorBoundary)).toHaveLength(1);
  });

  test('Should render Switch', () => {
    expect(Wrapper.find(Switch)).toHaveLength(1);
  });
});