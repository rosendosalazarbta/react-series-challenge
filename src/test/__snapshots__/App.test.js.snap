// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
>
        <styled.header>
                <styled.div>
                        <ul>
                                <styled.div>
                                        <Route
                                                path="/Home"
                                        >
                                                [Function]
                                        </Route>
                                </styled.div>
                                <styled.div>
                                        <Route
                                                path="/Favorites"
                                        >
                                                [Function]
                                        </Route>
                                </styled.div>
                        </ul>
                </styled.div>
        </styled.header>
        <styled.div>
                <ErrorBoundary>
                        <Switch>
                                <Route
                                        component={[Function]}
                                        path="/Home"
                                />
                                <Route
                                        component={[Function]}
                                        path="/Favorites"
                                />
                                <Redirect
                                        push={false}
                                        to={
                                                Object {
                                                        "pathname": "/Home",
                                                      }
                                        }
                                />
                        </Switch>
                </ErrorBoundary>
        </styled.div>
</div>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSaga": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": undefined,
          "done": Promise {},
          "error": [Function],
          "id": 1,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRunning": [Function],
          "joiners": Array [],
          "name": "root",
          "result": [Function],
          "setContext": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.header>
            <styled.div>
                        <ul>
                                    <styled.div>
                                                <Route
                                                            path="/Home"
                                                >
                                                            [Function]
                                                </Route>
                                    </styled.div>
                                    <styled.div>
                                                <Route
                                                            path="/Favorites"
                                                >
                                                            [Function]
                                                </Route>
                                    </styled.div>
                        </ul>
            </styled.div>
</styled.header>,
          <styled.div>
            <ErrorBoundary>
                        <Switch>
                                    <Route
                                                component={[Function]}
                                                path="/Home"
                                    />
                                    <Route
                                                component={[Function]}
                                                path="/Favorites"
                                    />
                                    <Redirect
                                                push={false}
                                                to={
                                                            Object {
                                                                        "pathname": "/Home",
                                                                      }
                                                }
                                    />
                        </Switch>
            </ErrorBoundary>
</styled.div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              <ul>
                            <styled.div>
                                          <Route
                                                        path="/Home"
                                          >
                                                        [Function]
                                          </Route>
                            </styled.div>
                            <styled.div>
                                          <Route
                                                        path="/Favorites"
                                          >
                                                        [Function]
                                          </Route>
                            </styled.div>
              </ul>
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ul>
                <styled.div>
                                <Route
                                                path="/Home"
                                >
                                                [Function]
                                </Route>
                </styled.div>
                <styled.div>
                                <Route
                                                path="/Favorites"
                                >
                                                [Function]
                                </Route>
                </styled.div>
</ul>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <styled.div>
                    <Route
                                        path="/Home"
                    >
                                        [Function]
                    </Route>
</styled.div>,
                  <styled.div>
                    <Route
                                        path="/Favorites"
                    >
                                        [Function]
                    </Route>
</styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Route
                      path="/Home"
>
                      [Function]
</Route>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": [Function],
                      "path": "/Home",
                    },
                    "ref": null,
                    "rendered": [Function],
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Route
                      path="/Favorites"
>
                      [Function]
</Route>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": [Function],
                      "path": "/Favorites",
                    },
                    "ref": null,
                    "rendered": [Function],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ErrorBoundary>
              <Switch>
                            <Route
                                          component={[Function]}
                                          path="/Home"
                            />
                            <Route
                                          component={[Function]}
                                          path="/Favorites"
                            />
                            <Redirect
                                          push={false}
                                          to={
                                                        Object {
                                                                      "pathname": "/Home",
                                                                    }
                                          }
                            />
              </Switch>
</ErrorBoundary>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Switch>
                <Route
                                component={[Function]}
                                path="/Home"
                />
                <Route
                                component={[Function]}
                                path="/Favorites"
                />
                <Redirect
                                push={false}
                                to={
                                                Object {
                                                                "pathname": "/Home",
                                                              }
                                }
                />
</Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    path="/Home"
/>,
                  <Route
                    component={[Function]}
                    path="/Favorites"
/>,
                  <Redirect
                    push={false}
                    to={
                                        Object {
                                                            "pathname": "/Home",
                                                          }
                    }
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/Home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/Favorites",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "push": false,
                    "to": Object {
                      "pathname": "/Home",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
>
          <styled.header>
                    <styled.div>
                              <ul>
                                        <styled.div>
                                                  <Route
                                                            path="/Home"
                                                  >
                                                            [Function]
                                                  </Route>
                                        </styled.div>
                                        <styled.div>
                                                  <Route
                                                            path="/Favorites"
                                                  >
                                                            [Function]
                                                  </Route>
                                        </styled.div>
                              </ul>
                    </styled.div>
          </styled.header>
          <styled.div>
                    <ErrorBoundary>
                              <Switch>
                                        <Route
                                                  component={[Function]}
                                                  path="/Home"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  path="/Favorites"
                                        />
                                        <Redirect
                                                  push={false}
                                                  to={
                                                            Object {
                                                                      "pathname": "/Home",
                                                                    }
                                                  }
                                        />
                              </Switch>
                    </ErrorBoundary>
          </styled.div>
</div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "runSaga": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": Array [],
            "name": "root",
            "result": [Function],
            "setContext": [Function],
          },
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.header>
              <styled.div>
                            <ul>
                                          <styled.div>
                                                        <Route
                                                                      path="/Home"
                                                        >
                                                                      [Function]
                                                        </Route>
                                          </styled.div>
                                          <styled.div>
                                                        <Route
                                                                      path="/Favorites"
                                                        >
                                                                      [Function]
                                                        </Route>
                                          </styled.div>
                            </ul>
              </styled.div>
</styled.header>,
            <styled.div>
              <ErrorBoundary>
                            <Switch>
                                          <Route
                                                        component={[Function]}
                                                        path="/Home"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        path="/Favorites"
                                          />
                                          <Redirect
                                                        push={false}
                                                        to={
                                                                      Object {
                                                                                    "pathname": "/Home",
                                                                                  }
                                                        }
                                          />
                            </Switch>
              </ErrorBoundary>
</styled.div>,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.div>
                <ul>
                                <styled.div>
                                                <Route
                                                                path="/Home"
                                                >
                                                                [Function]
                                                </Route>
                                </styled.div>
                                <styled.div>
                                                <Route
                                                                path="/Favorites"
                                                >
                                                                [Function]
                                                </Route>
                                </styled.div>
                </ul>
</styled.div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ul>
                  <styled.div>
                                    <Route
                                                      path="/Home"
                                    >
                                                      [Function]
                                    </Route>
                  </styled.div>
                  <styled.div>
                                    <Route
                                                      path="/Favorites"
                                    >
                                                      [Function]
                                    </Route>
                  </styled.div>
</ul>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <styled.div>
                      <Route
                                            path="/Home"
                      >
                                            [Function]
                      </Route>
</styled.div>,
                    <styled.div>
                      <Route
                                            path="/Favorites"
                      >
                                            [Function]
                      </Route>
</styled.div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Route
                        path="/Home"
>
                        [Function]
</Route>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": [Function],
                        "path": "/Home",
                      },
                      "ref": null,
                      "rendered": [Function],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Route
                        path="/Favorites"
>
                        [Function]
</Route>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": [Function],
                        "path": "/Favorites",
                      },
                      "ref": null,
                      "rendered": [Function],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "ul",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <ErrorBoundary>
                <Switch>
                                <Route
                                                component={[Function]}
                                                path="/Home"
                                />
                                <Route
                                                component={[Function]}
                                                path="/Favorites"
                                />
                                <Redirect
                                                push={false}
                                                to={
                                                                Object {
                                                                                "pathname": "/Home",
                                                                              }
                                                }
                                />
                </Switch>
</ErrorBoundary>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Switch>
                  <Route
                                    component={[Function]}
                                    path="/Home"
                  />
                  <Route
                                    component={[Function]}
                                    path="/Favorites"
                  />
                  <Redirect
                                    push={false}
                                    to={
                                                      Object {
                                                                        "pathname": "/Home",
                                                                      }
                                    }
                  />
</Switch>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      path="/Home"
/>,
                    <Route
                      component={[Function]}
                      path="/Favorites"
/>,
                    <Redirect
                      push={false}
                      to={
                                            Object {
                                                                  "pathname": "/Home",
                                                                }
                      }
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/Home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "path": "/Favorites",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "push": false,
                      "to": Object {
                        "pathname": "/Home",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
